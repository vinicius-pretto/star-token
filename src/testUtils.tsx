import * as React from "react";
import { render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore, { initialState } from "./ui/redux/store";

const Providers = ({ children, state }) => {
  return (
    <ReduxProvider store={configureStore(state)}>{children}</ReduxProvider>
  );
};

const customRender = (ui, state = initialState) => {
  const options = {
    wrapper: (props) => <Providers {...props} state={state} />,
  };
  return render(ui, options);
};

export * from "@testing-library/react";
export { customRender as render };
