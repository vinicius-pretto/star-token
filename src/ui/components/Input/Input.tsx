import * as React from "react";

interface Props {
  id: string;
  type: string;
  value: string;
  onChange: any;
  onBlur?: any;
  placeholder?: string;
  error?: any;
  touched?: any;
  maxLength?: number;
}

const Input = ({
  id,
  type,
  value,
  onChange,
  onBlur,
  placeholder,
  error,
  touched,
  maxLength,
}: Props) => {
  const hasError = touched && error;

  return (
    <div className="input mb-sm form-group">
      <input
        id={id}
        type={type}
        className="input__field"
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        placeholder={placeholder}
        maxLength={maxLength}
      />
      <p className={hasError ? "error" : "visibility-hidden"}>{error}</p>
    </div>
  );
};

export default Input;
