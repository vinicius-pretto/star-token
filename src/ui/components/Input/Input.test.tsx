import * as React from "react";
import { render } from "../../../testUtils";
import Input from "./Input";

describe("Input", () => {
  it("snapshot", () => {
    const { asFragment } = render(
      <Input
        id="name"
        type="text"
        value="#cc0000"
        onChange={jest.fn()}
        onBlur={jest.fn()}
        placeholder="color-primary"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
