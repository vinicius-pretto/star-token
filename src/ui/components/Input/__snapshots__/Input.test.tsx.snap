// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<DocumentFragment>
  <div
    class="input mb-sm form-group"
  >
    <input
      class="input__field"
      id="name"
      placeholder="color-primary"
      type="text"
      value="#cc0000"
    />
    <p
      class="visibility-hidden"
    />
  </div>
</DocumentFragment>
`;
